### PR TITLE
Release v0.9.0

### PR DESCRIPTION
**Pull request by fredcarle**
Release branch as of 58dc727a5e12ebecdd0adadec5188bd4b2a6697a

Release tasks completed:
- [x] ran `go mod tidy`
- [x] regenerated in-repo docs (CLI and manpages)
- [x] bump BSL license parameters
- [x] update changelog

Only need to review the last commit :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,82 @@
...
+
+* Add strong typing to document creation ([#2161](https://github.com/sourcenetwork/defradb/issues/2161))
+* Rename key,id,dockey to docID terminology ([#1749](https://github.com/sourcenetwork/defradb/issues/1749))
+* Simplify Merkle CRDT workflow ([#2111](https://github.com/sourcenetwork/defradb/issues/2111))
+
+### Testing
+
+* Add auto-doc generation ([#2051](https://github.com/sourcenetwork/defradb/issues/2051))
+
+### Continuous integration
```
Comment (AndrewSisley) [inline CHANGELOG.md:41]: thought: Do we want this header in the change log?

**Review by AndrewSisley — APPROVED** [review]
LGTM - thanks Fred!

File: CHANGELOG.md (hunk 1 above)
Comment (fredcarle) [reply CHANGELOG.md:39]: They were there in the past. It is not super relevant information for users so I'm open to removing it.

Comment (fredcarle) [reply CHANGELOG.md:39]: Lets keep it there for this one and discuss for future releases.

**Pull request merged**